### ISSUE TITLE
Websites using SSAI (server side ad injection)

### ISSUE THREAD

**Issue by ZaphodBeebblebrox** [uBlockOrigin/uAssets#1905]
### URL(s) where the issue occurs

`http://www.cwtv.com/shows/arrow/brothers-in-arms/?play=498cfef8-fa09-4b70-b860-88fd4f1e3362&extended`

### Describe the issue

At certain points in the video, ads play. They are marked on the bottom bar by the thicker sections. The first one starts at 7:45.

### Versions

- 65.0.3325.181 (Official Build) (64-bit)
- uBlock Origin version: 1.15.18

### Settings

Default settings

**Comment by mapx-** [comment]
as I can see it's about uplynk streaming (SSAI), so ... it seems nothing to do
https://github.com/uBlockOrigin/uAssets/issues/772#issuecomment-351716738

**Comment by ghajini** [comment]
Something like this 
`https://github.com/uBlockOrigin/uAssets/issues/1837`
Can help? Noopmp4

**Comment by mapx-** [comment]
an idea from ABP:
https://issues.adblockplus.org/ticket/6592

**Comment by uBlock-user** [comment]
Isn't that the same as $redirect directive ?

**Comment by gorhill** [comment]
They suggest to rewrite the URL without blocking the request -- `redirect`'s purpose is differen, and two-fold: 1st to block, then to redirect to a local resource such that no network request leaves the browser.

**Comment by gorhill** [comment]
> Can it be done? Noopmp4 @gorhill

So far I haven't been able to investigate any of the cases, they all require to create an account, and it seems one need to be in the US. For example, the URL here I get "Access forbidden" and the page renders badly.

**Comment by uBlock-user** [comment]
> the URL here I get "Access forbidden" and the page renders badly.

Same here, they purposefully **deface** the page upon detecting the IP being outside of the US.

**Comment by ghajini** [comment]
Not reproducing currently....
USA VPN
With nano defender

**Comment by gorhill** [comment]
The rewrite option is now available in ABP dev build. Does it work to fix the issue here?

**Comment by uBlock-user** [comment]
As per the [comment#11](https://issues.adblockplus.org/ticket/6592#comment:11) it does.

**Comment by gorhill** [comment]
That comment is for `http://6abc.com/shots-fired-at-police-in-gloucester-twp-standoff-schools-closed/2927128/`, not for the case here, `http://www.cwtv.com/shows/arrow/brothers-in-arms/?play=498cfef8-fa09-4b70-b860-88fd4f1e3362&extended`. I am asking for a verified answer, not a speculated one, i.e. is it working for the case here and with which filters exactly. I can't test this myself, I get a "Access is forbidden" for the URL in opening comment.

**Comment by uBlock-user** [comment]
> That comment is for http://6abc.com/shots-fired-at-police-in-gloucester-twp-standoff-schools-closed/2927128/, not for the case here, http://www.cwtv.com/shows/arrow/brothers-in-arms/?play=498cfef8-fa09-4b70-b860-88fd4f1e3362&extended

They both use [uplynk](https://issues.adblockplus.org/ticket/6242#comment:7), wouldn't that make them both the same test case ?

**Comment by zaphodbeebblebrox** [comment]
Strangely, they appear to have stopped using uplynk. At the moment, I cannot find something that uses the uplynk player, and ads are skipped. I guess this issue should be closed? (It doesn't really feel like my issue, so I shall leave it up to others discretion.)

**Comment by mapx-** [comment]
6abc still using uplynk, and $rewrite filter is working only for chrome, firefox getting a black screen video (as reported by on ABP issue). Probably it's the same issue uBo has with xhr redirections on FF.

**Comment by uBlock-user** [comment]
They saw ABP's new directive and reacted, they will come up with something else. Don't expect it to be one and done.

**Comment by okiehsch** [comment]
These websites are using SSAI
````
6abc.com, abc7.com, cwtv.com, fox.com, nbc.com, viceland.com
````

**Comment by jspenguin2017** [comment]
I got one more report, it has New Zealand GeoLock and requires account. From network request, it requests manifest from Unicorn Media (owned by Brightcove), so I assume it is server side ads injection as well. 
```
https://www.threenow.co.nz/shows/svu/season-19-ep-20/S1543-013/M22244-050
```

**Comment by ZaphodBeebblebrox** [comment]
More:
```
www.discovery.com 
www.sciencechannel.com
```

**Comment by Alanimus** [comment]
This is a real problem. I spent a couple hours trying out the URL rewrite with adblock plus and while the filter works, I can't block the ads. Could someone explain how this should work? You can bypass the geo block with VPN and the "unlocked" videos have the same problem on investigationdiscovery.com so you don't need a TV account or anything.

**Comment by jspenguin2017** [comment]
They saw ABP devs playing around with the new filter syntax and updated their system to break the filter.

**Comment by Alanimus** [comment]
Ha, figures. Man that ticks me off. It doesn't matter if you pay for it or
not (I have a cable TV account) you are still inundated with up to 7 ads at
once. All this does is discourage people from watching it through legal
means.

**Comment by terrorist96** [comment]
`https://abc7ny.com/plane-with-blown-tires-attempting-to-land-at-nj-airport/4019303/`

**Comment by mapx-** [comment]
https://github.com/uBlockOrigin/uAssets/issues/2384
https://7plus.com.au/AST?episode-id=AST-245

**Comment by uBlock-user** [comment]
https://github.com/uBlockOrigin/uAssets/issues/1173 too

**Comment by terrorist96** [comment]
`abc13.com`

**Comment by krystian3w** [comment]
`thecwtv.com`

**Comment by terrorist96** [comment]
`https://abc7chicago.com/woman-who-was-pinned-by-suv-at-indiana-toll-booth-dies-four-days-later/4904557/`

**Comment by THEtomaso** [comment]
Looks like you can add `nettavisen.no` to this list as well:
https://github.com/DandelionSprout/adfilt/issues/7#issuecomment-449497812

**Comment by DandelionSprout** [comment]
Note that the inclusion of SSAI on `nettavisen.no` doesn't cover *all* videos on the site, and covers virtually none of the non-geoblocked videos. More details at https://github.com/DandelionSprout/adfilt/issues/7#issuecomment-449560661.

**Comment by okiehsch** [comment]
As mentioned at https://github.com/DandelionSprout/adfilt/issues/7#issuecomment-449569146
````
nettavisen.no##.vjs-ssai-ad-click-target
nettavisen.no##.vjs-hidden:style(display: initial !important;)
nettavisen.no##.vjs-hidden:not(.vjs-ssai-skip-ad-btn)
````
enables the skip ad button which would be atleast a partial fix, so maybe that also works for other sites using SSAI.

**Comment by jasonkingsley** [comment]
### URL(s) where the issue occurs

`https://www.investigationdiscovery.com/tv-shows/true-crime-with-aphrodite-jones`

### Describe the issue

Video ads are no longer blocked on this site. They haven't been for the last 6-7 months.

### Screenshot(s)

https://i.imgur.com/XQkLRBV.jpg

### Versions

- Browser/version: Chrome - Version 71.0.3578.98 
- uBlock Origin version: uBlock Origin v1.17.4

### Settings

No changes made to default settings.

### Notes

Ublock Inspector screenshot:

Each time video ads start playing it shows 3 of those blocked scorecard URLs in a row. Same for each of the individual video ads once the ads start playing. It also shows the 3 blocked URL ads right before the show resumes playing.

**Comment by 241d2** [comment]
For `discovery.com` and maybe all discovery network sites, I used SVP4 Pro (which fakes 60fps and it's looking really great for 30fps videos) to grab m3u8 and play it directly in MPV player. No ads in it. It can be used as a workaround

If you really need workaround, maybe you can play directly on MPV player official version like `mpv.exe https://www.discovery.com/tv-shows/moonshiners/full-episodes/legacy-on-the-line` with youtube-dl.exe placed next to it

**Comment by okiehsch** [comment]
https://github.com/uBlockOrigin/uAssets/issues/5803

**Comment by llacb47** [comment]
Can someone test this filter `discovery.com##+js(set-constant.js, vpaidAdUnit.initAd, noopFunc)`?

It skips some midroll ads (not all of them are injection).

Test URL: `https://go.discovery.com/tv-shows/shark-week/full-episodes/air-jaws-strikes-back` (US geolock?)

**Comment by terrorist96** [comment]
> discovery.com##+js(set-constant.js, vpaidAdUnit.initAd, noopFunc)

I still see ads.

**Comment by llacb47** [comment]
Hmm

**Comment by llacb47** [comment]
I will try to take a video recording of this filter in https://github.com/uBlockOrigin/uAssets/issues/1905#issuecomment-517960703 in action to show that it works.

**Comment by llacb47** [comment]
@okiehsch please replace 
```
@@||fusion.ddmcdn.com^$script,domain=discovery.com|investigationdiscovery.com|sciencechannel.com
@@||src.litix.io/core/*/mux.js$script,domain=discovery.com|investigationdiscovery.com|sciencechannel.com
```
with
```
@@||fusion.ddmcdn.com^$script,domain=animalplanet.com|cookingchanneltv.com|discovery.com|diynetwork.com|foodnetwork.com|hgtv.com|investigationdiscovery.com|sciencechannel.com|tlc.com|travelchannel.com
@@||src.litix.io/core/*/mux.js$script,domain=animalplanet.com|cookingchanneltv.com|discovery.com|diynetwork.com|foodnetwork.com|hgtv.com|investigationdiscovery.com|sciencechannel.com|tlc.com|travelchannel.com
```

(in the unbreak list)

**Comment by prateekrastogi** [comment]
Despite multiple issues to fix Bing Ads, there has been no response whatsoever and agitation from repo maintainers:
So, referencing my issue here:
https://github.com/uBlockOrigin/uAssets/issues/6887#issuecomment-578509029

Let's name and shame ublock origin until their ego issues get resolved.

**Comment by terrorist96** [comment]
`https://abc11.com/politics/another-nc-county-passes-2nd-amendment-sanctuary-measure/5869857/`

**Comment by uBlock-user** [comment]
`https://abc13.com/traffic/body-found-during-search-for-crosby-firefighter-thrown-from-jeep/6250058/`

**Comment by PIZZA90** [comment]
```
https://www.kissfm.es/player/
https://play.los40.com/
```
audio **SSAI** ads??

**Comment by llacb47** [comment]
Yes, most radio streams have instream commercials.

**Comment by THEtomaso** [comment]
### Possible fix for Sky.com:
```
@@||ping.chartbeat.net^$domain=sky.com
@@||static.chartbeat.com/js/chartbeat.js$domain=sky.com
```
At least, with these whitelistings in place, the videos will play, after refreshing the page a couple of times.
Considering the large amount of problem filters (se below), **uBlock filters - Unbreak** seems like a natural place for a potential fix, unless a more clever solution can be conjured up for **uBlock filters**.

### Example:
`https://news.sky.com/video/huge-shark-swims-near-family-boat-in-nova-scotia-12037282/`

### Problem filters:
* **AdGuard Tracking Protection filter** (@BlazDT):
`||chartbeat.com^$third-party`

* **Dan Pollock's hosts file**:
```
chartbeat.com
ping.chartbeat.net
static.chartbeat.com
```

* **EasyPrivacy** (@Khrin):
```
/chartbeat.js
||chartbeat.com^$third-party
```

* **hpHosts - Ad and Tracking servers only** [DISCONTINUED]:
```
ping.chartbeat.net
static.chartbeat.com
```

* **Lunar Blocklist by Moonchild** (@wolfbeast):
`||chartbeat.`

* **Malvertising list by Disconnect**:
`chartbeat.com`

* **MVPS HOSTS**:
`ping.chartbeat.net`

* **Peter Lowe's Ad and tracking server list** (@pgl):
`chartbeat.com`

* **Schacks Adblock Plus liste** [DISCONTINUED]:
```
||ping.chartbeat.net
||static.chartbeat.com
```

--

### System info:
* OS/version: Windows 8.1 Pro (x64)
* Browser/version: Pale Moon v28.11.0 (x64)
* Adblock Extension/version: uBlock Origin v1.16.4.24 (XUL)

**Comment by BlazDT** [comment]
Sometimes the video works for me, sometimes not (even when filters are not enabled). Please check @AdamWr

**Comment by AdamWr** [comment]
Regarding - https://github.com/uBlockOrigin/uAssets/issues/1905#issuecomment-664468079
At the first visit this video doesn't work even without any ad blocker on my side, but after refreshing website, it works even if `chartbeat.com` is blocked.

**Comment by THEtomaso** [comment]
If you've allowed the connections once, then the videos will play afterwards, event if `chartbeat.com` is blocked.
But if you start fresh, with an empty cache, the videos won't play if `chartbeat.com` is blocked, no matter how many times you refresh the page!
At least, this is what happens at my end.

**Comment by AdamWr** [comment]
I checked it two times in incognito/private tab.
First time with ad blocker enabled and at the first visit video didn't work, but after refreshing website it worked fine and `chartbeat.com` was blocked.
Second time I checked it without any ad blocker with the same result, but of course `chartbeat.com` wasn't blocked.

**Comment by THEtomaso** [comment]
I managed to get it to play that way too now, after many page refreshes.
I hate it, when this type of inconsistent behaviour manifests itself, because it makes things very difficult to troubleshoot!
Nevertheless, since we actually ARE able to play the videos without ads (at least in this case)..
then perhaps the issue can be remedied, through the use of some type of clever `##+js` fix (despite that `fox.com` is listed under this "can't fix" report)?

**Comment by Lv44EnderShaman** [comment]
I'm a bit confused by this. To sum it up, is SSAI basically unblockable by our currently filtering tools? I don't understand the context, here.

**Comment by uBlock-user** [comment]
abc7ny -- `https://abc7ny.com/stabbing-outside-hotel-robbery-four-points-nyc-crime/6421164/` reported at `https://www.reddit.com/r/uBlockOrigin/comments/iswbej/how_to_remove_video_ads_on/`

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uAssets/issues/7968

**Comment by krystian3w** [comment]
#5184

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uAssets/issues/7968 https://github.com/uBlockOrigin/uAssets/issues/5184 https://github.com/uBlockOrigin/uAssets/issues/1905

**Comment by gorhill** [comment]
Using `queryprune`, I was able to skip ads at:

    https://6abc.com/shots-fired-at-police-in-gloucester-twp-standoff-schools-closed/2927128/

Using:

    ||content.uplynk.com/api/*&ad=$xhr,queryprune=|ad=,domain=6abc.com

So we will have to go through those cases above and see which ones are now fixable.

It's ok to start to add `queryprune` to filter lists, the filters will just be discarded as invalid on older versions of uBO.

**Comment by uBlock-user** [comment]
See - https://github.com/uBlockOrigin/uAssets/issues/9764

**Comment by mapx-** [comment]
`https://www.discoveryplus.com/it/video/fratelli-di-crozza/stagione-5-puntata-24-del-10-dicembre-2021`

https://github.com/uBlockOrigin/uAssets/issues/10870

**Comment by krystian3w** [comment]
tested `queryprune`?

**Comment by mapx-** [comment]
https://github.com/uBlockOrigin/uAssets/issues/11302

`travelchannel.com`
free episode
`https://watch.travelchannel.com/video/ghost-adventures-travel-channel/horror-in-biggs`

**Comment by uBlock-user** [comment]
another - `https://freeworldnews.tv/watch?id=6203cc6da3107002d4231671`

**Comment by fedepr** [comment]
hello, i'm a bit confused, is there a way , actually, to see discovery plus without ads (e.g. https://www.discoveryplus.com/it/video/primo-appuntamento/stagione-6-episodio-1) ?
Thanks!!

**Comment by mapx-** [comment]
No, nothing to do for now.

**Comment by bandaid** [comment]
> ### URL(s) where the issue occurs
> 
> `http://www.cwtv.com/shows/arrow/brothers-in-arms/?play=498cfef8-fa09-4b70-b860-88fd4f1e3362&extended`
> ### Describe the issue
> 
> At certain points in the video, ads play. They are marked on the bottom bar by the thicker sections. The first one starts at 7:45.
> ### Versions
> 
>     * 65.0.3325.181 (Official Build) (64-bit)
> 
>     * uBlock Origin version: 1.15.18
> 
> 
> ### Settings
> 
> Default settings

I never used the site but it is broken for me. Maybe the site is dead.

![Screenshot 2022-03-03 at 22-42-58 The CW Network Official Site The CW CW Seed](https://user-images.githubusercontent.com/8212255/156665461-f5418f27-a5b7-497d-9728-9c1df8b0fa10.png)

**Comment by krystian3w** [comment]
For only you - links only redirect to 404 `https://www.cwtv.com/page-not-found/` and CW are not bankrupt.

![obraz](https://user-images.githubusercontent.com/35370833/156667486-ce34c830-22fe-43a8-a70c-571f5a1b3c46.png)

So rendered correctly.

**Comment by Prajwal-Koirala** [comment]
`nbc.com` is using this too, need a fix for this.

**Comment by krystian3w** [comment]
If possible with removeparam.

**Comment by mapx-** [comment]
https://github.com/uBlockOrigin/uAssets/issues/12407

**Comment by mapx-** [comment]
`philo.com`
https://github.com/uBlockOrigin/uAssets/issues/12469

**Comment by mapx-** [comment]
https://github.com/uBlockOrigin/uAssets/issues/12513

**Comment by stephenhawk8054** [comment]
`mech-plus.com`
[https://github.com/uBlockOrigin/uAssets/issues/12589](https://github.com/uBlockOrigin/uAssets/issues/12589)

**Comment by stephenhawk8054** [comment]
`amc.com`
[https://github.com/uBlockOrigin/uAssets/issues/13065](https://github.com/uBlockOrigin/uAssets/issues/13065)

**Comment by rated9** [comment]
anyone know any adblock that blocks atleast one of these websites (https://www.investigationdiscovery.com/) or tlc/ ahc /  travel channel / destination america/ i tried alot of adblockers none working :(

**Comment by krystian3w** [comment]
Use one at time, to combo...

**Comment by rated9** [comment]
> Use one at time, not combo...

what u mean ?

**Comment by rated9** [comment]
look in the image bellow sometimes i manage to bug the player by doing some trick and than i can watch anyvideo without any ads but if i close the browser or turn off vpn than its back with ads 
![image](https://user-images.githubusercontent.com/106842792/171962276-674c0830-f6b8-480d-9d34-4e1e5c10e075.png)

**Comment by stephenhawk8054** [comment]
`vix.com`
https://github.com/uBlockOrigin/uAssets/issues/14213

**Comment by Infrastructural** [comment]
https://github.com/uBlockOrigin/uAssets/issues/14535
https://github.com/uBlockOrigin/uAssets/commit/468574689335cac40354dc8511fa1242ed2ad4af

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uAssets/issues/14535

**Comment by snvoid** [comment]
is Crackle using SSAI now? I seem to be getting ads

**Comment by peace2000** [comment]
What's the current situation, is there anything that can be done to `discoveryplus.com` ?

This is trailer which doesnt' require an account to watch. It shows an ad too.

`https://www.discoveryplus.com/fi/video/atlantin-yli/traileri-kausi-3`

**Comment by stephenhawk8054** [comment]
@peace2000 Does this work on your side?

```adb
discoveryplus.com##+js(json-prune, data.attributes.ssaiInfo.forecastTimeline data.attributes.ssaiInfo.vendorAttributes.nonLinearAds data.attributes.ssaiInfo.vendorAttributes.videoView)
discoveryplus.com##+js(xml-prune, Period:not([duration*="M"]), , .mpd)
```

**Comment by peace2000** [comment]
@stephenhawk8054 Sorry I was slow to respond :) Yeah seems to work just fine.

Also works for normal episode: `https://www.discoveryplus.com/fi/video/wahlgrens-varld/stor-familjemiddag-hos-wahlgrens`.

(Free, but requires an account).

**Comment by terrorist96** [comment]
I saw that this longstanding issue was closed. I checked to test to see if it was really fixed and most seem to be fixed. But I found one that still has ads:
`https://go.discovery.com/video/expedition-unknown-discovery-atve-us/amelia-earhart`
Seeing an ad at the beginning. Might have ads throughout as well. I didn't watch long enough to test. Didn't want to make a new issue to avoid duplicates. @mapx-

**Comment by peace2000** [comment]
@terrorist96 Please open a report here: https://github.com/uBlockOrigin/uAssets/issues
